Public/Src/Cache/ContentStore/Distributed/NuCache/LocalLocationStoreConfiguration.cs

```diff
@@ -203,32 +205,87 @@ public DistributedCentralStoreConfiguration(AbsolutePath cacheRoot)
         public int MaxSimultaneousCopies { get; set; } = 10;
     }
 
+    /// <summary>
+    /// Provides Azure Storage authentication options for <see cref="BlobCentralStorage"/>
+    /// </summary>
+    public class AzureBlobStorageCredentials
+    {
+        /// <nodoc />
+        public readonly string ConnectionString;
+
+        /// <summary>
+        /// <see cref="StorageCredentials"/> can be updated from the outside, so it is a way to in fact change the way
+        /// we authenticate with Azure Blob Storage over time. Changes are accepted only within the same authentication
+        /// mode.
+        /// </summary>
+        public readonly StorageCredentials StorageCredentials;
+
+        /// <nodoc />
+        public readonly string AccountName;
+
+        /// <nodoc />
+        public readonly string EndpointSuffix;
+
+        /// <summary>
+        /// Creates a fixed credential; this is our default mode of authentication.
+        /// </summary>
+        public AzureBlobStorageCredentials(string connectionString)
+        {
+            Contract.Requires(!string.IsNullOrEmpty(connectionString));
+            ConnectionString = connectionString;
+        }
+
+        /// <summary>
+        /// Uses Azure Blob's storage credentials. This allows us to use SAS tokens, and to update shared secrets
+        /// without restarting the service.
+        /// </summary>
+        public AzureBlobStorageCredentials(StorageCredentials storageCredentials, string accountName, string endpointSuffix)
+        {
+            // Unfortunately, even though you can't generate a storage credentials without an account name and
+            // endpoint suffix, those aren't stored inside object unless a shared secret is being used. Hence, we are
+            // forced to keep those here as well.
+            Contract.Requires(storageCredentials != null);
+            Contract.Requires(!string.IsNullOrEmpty(accountName));
+            Contract.Requires(!string.IsNullOrEmpty(endpointSuffix));
+            StorageCredentials = storageCredentials;
+            AccountName = accountName;
+            EndpointSuffix = endpointSuffix;
+        }
+    }
+
     /// <summary>
     /// Configuration of a central store backed by azure blob storage.
     /// </summary>
     public class BlobCentralStoreConfiguration : CentralStoreConfiguration

```

Comment:
I actually was wrong, and the entire code that deals with this class is in this codebase not in Cloudbuild.

You can change DistributedContentStoreFactory type from BuildXL.Cache.Host.SErvice to use your new api.